packages/shared/src/components/ProfileMenu.tsx

```diff
@@ -35,85 +41,95 @@ export default function ProfileMenu(): ReactElement {
+        target_id: TargetId.InviteProfileMenu,
+      });
+      openModal({ type: LazyModal.SearchReferral });
+    };
+
+    const list: ListItem[] = [
+      {
+        title: 'Profile',
+        buttonProps: {
+          tag: 'a',
+          href: user.permalink,
+          icon: <UserIcon />,
+        },
+      },
+      {
+        title: 'Account details',
+        buttonProps: {
+          tag: 'a',
+          icon: <SettingsIcon />,
+          href: `${webappUrl}account/profile`,
+        },
+      },
+      {
+        title: 'Devcard',
+        buttonProps: {
+          tag: 'a',
+          icon: <DevCardIcon />,
+          href: `${webappUrl}devcard`,
+        },
+      },
+    ];
+
+    if (showSearchReferral) {
+      list.push({
+        title: `${availableCount} Search invites`,
+        buttonProps: {
+          icon: <KeyReferralOutlineIcon secondary />,
+          onClick: handleReferralClick,
+        },
+      });
+    }
+
+    list.push({
+      title: 'Logout',
+      buttonProps: {
+        icon: <PowerIcon />,
+        onClick: logout,
+      },
+    });
+
+    return list;
+  }, [trackEvent, logout, availableCount, openModal, showSearchReferral, user]);
+
   if (!user) {
     return <></>;
   }
 
-  const handleReferralClick = () => {
-    trackEvent({
-      event_name: AnalyticsEvent.Click,
-      target_type: TargetType.SearchInviteButton,
-      target_id: TargetId.InviteProfileMenu,
-    });
-    openModal({ type: LazyModal.SearchReferral });
-  };
-
   return (
-    <PortalMenu
-      disableBoundariesCheck
-      id="profile-context"
-      className="menu-primary"
-      animation="fade"
+    <InteractivePopup
+      onClose={onClose}
+      closeOutsideClick
+      position={InteractivePopupPosition.RightStart}
+      className="w-full border !right-4 laptop:max-w-[13.75rem] max-w-[21.25rem] roundeed-14 border-theme-divider-tertiary"
     >
-      <Item>
-        <Link href={user.permalink} passHref prefetch={false}>
-          <a className="flex items-center w-full">
-            <UserIcon
-              size={IconSize.Small}
-              secondary={false}
-              className="mr-2"
-            />{' '}
-            Profile
-          </a>
-        </Link>
-      </Item>
-      <Item>
-        <Link
-          href={`${process.env.NEXT_PUBLIC_WEBAPP_URL}account/profile`}
-          passHref
-          prefetch={false}
-        >
-          <a className="flex items-center w-full">
-            <SettingsIcon
-              size={IconSize.Small}
-              secondary={false}
-              className="mr-2"
-            />{' '}
-            Account details
-          </a>
-        </Link>
-      </Item>
-      {showSearchReferral && (
-        <Item>
-          <button
-            type="button"
-            className="flex items-center min-w-[12.5rem]"
-            onClick={handleReferralClick}
+      <LabeledImage
+        src={user.image}
+        alt="Logged-in user"
```

Comment:
Our existing implementations, iirc, contain the word image/avatar/photo, which linting does not accept, we only get away with it because we pass it through a reusable component which lint won't know we will use the alt for the `img` element. Also, here, we don't need to know which specific user is logged in. Wdyt?

Reply:
The component is only for logged users no?
I prefer to have it more explicit as usernames photo but it's a small semantic

Reply:
Yeah, that makes sense. Explicit is always better.